src/main/java/io/nats/client/impl/MessageQueue.java

```diff
@@ -49,7 +50,9 @@
      * @param publishHighwaterMark sets a limit on the size of the underlying queue
      */
     MessageQueue(boolean singleReaderMode, int publishHighwaterMark) {
-        this.queue = publishHighwaterMark > 0 ? new LinkedBlockingQueue<NatsMessage>(publishHighwaterMark) : new LinkedBlockingQueue<NatsMessage>();
+        this.queue = new LinkedBlockingQueue<NatsMessage>();

```

Comment:
so what is publishHighwaterMark for?

Reply:
It looks like a publish buffer, thats set to infinite during reconnect.  I don't believe there's an analogous buffer in the other clients.